state/mongodb/mongodb.go

```diff
@@ -113,6 +113,19 @@ func (m *MongoDB) Init(metadata state.Metadata) error {
 
 // Set saves state into MongoDB
 func (m *MongoDB) Set(req *state.SetRequest) error {
+	ctx, cancel := context.WithTimeout(context.Background(), m.operationTimeout)
```

Comment:
JFYI - I'm worried that we use context.Background() in many places. In DotNet, cancellationtoken is used to cooperate cancellation among thread/thread pools/work items/Tasks and normally we pass this token down to function calls. 

In Go, Incoming request has its own context and its context should be properly propagated to function call chains during the request life-cycle.  I created the issue pointing out this - https://github.com/dapr/dapr/issues/292

Reply:
We do not need to handle this issue now :)

Reply:
For components specifically, in most cases contexts are created when a metadata property for a timeout is given. That means the context is created by a timeout defined by the user. If a transient connection occurs between the client and Dapr which the user didn't mean to cancel, it will cancel an underlying operation also even though the user did not mean to. That will hurt some of Dapr's reliability guarantees.